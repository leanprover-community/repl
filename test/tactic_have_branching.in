{"cmd": "example : 1 = 1 := by sorry", "sorries": "individual"}

{"tactic": "rfl", "proofState": 0}

{"tactic": "have : True := by sorry", "proofState": 0}

{"tactic": "exact h1", "proofState": 2}

{"tactic": "exact True.intro", "proofState": 2}

{"tactic": "rfl", "proofState": 4}

