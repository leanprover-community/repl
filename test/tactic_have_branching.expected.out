{"sorries":
 [{"proofState": 0,
   "pos": {"line": 1, "column": 22},
   "parentDecl": "_example",
   "goals": ["⊢ 1 = 1"],
   "endPos": {"line": 1, "column": 27}}],
 "messages":
 [{"severity": "warning",
   "pos": {"line": 1, "column": 0},
   "endPos": {"line": 1, "column": 7},
   "data": "declaration uses 'sorry'"}],
 "env": 0}

{"proofStatus": "Completed", "proofState": 1, "goals": []}

{"proofStatus": "Incomplete: open goals remain",
 "proofState": 2,
 "goals": ["⊢ True", "this : True\n⊢ 1 = 1"]}

{"proofStatus": "Incomplete: open goals remain",
 "proofState": 3,
 "messages":
 [{"severity": "error",
   "pos": {"line": 0, "column": 0},
   "endPos": {"line": 0, "column": 0},
   "data": "unknown identifier 'h1'"}],
 "goals": ["⊢ True", "this : True\n⊢ 1 = 1"]}

{"proofStatus": "Incomplete: open goals remain",
 "proofState": 4,
 "goals": ["this : True\n⊢ 1 = 1"]}

{"proofStatus": "Completed", "proofState": 5, "goals": []}

